src/pal/src/loader/module.cpp

```diff
@@ -840,6 +840,58 @@ PAL_GetSymbolModuleBase(void *symbol)
     return retval;
 }
 
+/*++
+    PAL_GetLoadLibraryError
+
+    Wrapper for dlerror() to be used by PAL functions
+
+Parameters:
+    IN nSize - the number of characters to copy to lpBuffer
+    OUT lpBuffer - the buffer to copy dlerror() to
+
+Return value:
+
+If the buffer pointed to by lpBuffer is not large enough, the return
+value is the buffer size required to hold the value string. Otherwise, return 0.
+
+--*/
+PALIMPORT
+DWORD
+PALAPI
+PAL_GetLoadLibraryError(
+    OUT LPWSTR lpBuffer,
+    IN DWORD nSize)
+{
+    DWORD retval = 0;
+    INT error_length;
+    LPWSTR last_error = nullptr;
+    
+    PERF_ENTRY(PAL_GetLoadLibraryError);
+    ENTRY("PAL_GetLoadLibraryError (lpBuffer=%p, nSize=%u)\n",
+          lpBuffer, nSize);
+
+    wcscpy_s(lpBuffer, nSize, W(""));
+
+    last_error = (LPWSTR) dlerror();
+    error_length = lstrlenW(last_error);
+
+    if (error_length >= (INT)nSize)
+    {
+        TRACE("Buffer too small (%u) to copy last error message (%u).\n", nSize, error_length);
```

Comment:
Hmm this is going to clear dlerror() so when you call again with the correct buffer size it gives nothing. We need a different scheme.

Reply:
@jkotas I assume the PAL cannot return memory it allocated itself? Is there a better scheme than just passing in a buffer that's hopefully large enough?

Reply:
The PAL can just return the pointer from dlerror. Without any copies, etc.

It may be nice to call the method `PAL_dlerror` to make it clear that it is just dlerror wrapper.